File: app/src/main/java/com/zeyad/rxredux/screens/user/list/events/GetPaginatedUsersEvent.java
package com.zeyad.rxredux.screens.user.list.events;

import com.zeyad.rxredux.core.redux.BaseEvent;

/**
 * @author by ZIaDo on 4/19/17.
 */
public class GetPaginatedUsersEvent implements BaseEvent<Long> {

    private final long lastId;

    public GetPaginatedUsersEvent(long lastId) {
        this.lastId = lastId;
    }

    @Override
    public Long getPayLoad() {
        return lastId;
    }
}

File: app/src/main/java/com/zeyad/rxredux/screens/user/list/events/GetPaginatedUsersEvent.kt
package com.zeyad.rxredux.screens.user.list.events

import com.zeyad.rxredux.core.BaseEvent
import kotlinx.android.parcel.Parcelize

@Parcelize
class GetPaginatedUsersEvent(private val lastId: Long) : BaseEvent<Long> {
    override fun getPayLoad(): Long = lastId
}
